sql/core/src/test/scala/org/apache/spark/sql/test/DataFrameReaderWriterSuite.scala

```diff
@@ -395,7 +395,7 @@ class DataFrameReaderWriterSuite extends QueryTest with SharedSparkSession with
       .load("/test")
 
     assert(LastOptions.parameters("intOpt") == "56")
-    assert(LastOptions.parameters("path") == "/test")
+    assert(!LastOptions.parameters.contains("path"))
```

Comment:
Sorry I missed one thing. For DS v1, `...load(path)` will generate a `path` option, which is visible to v1 sources, and we can't break it. It's strange that `...load(path1, path2)` will completely ignore the paths and don't pass paths to v1 sources, but there is nothing we can do here.

We need to update `DataFrameReader.loadV1Source` to generate a `path` option if `paths.length == 1`, to keep the previous behavior.

Reply:
Yes, now it makes sense. I will update it, thanks!